src/contraction/doc/contraction.rst

```diff
@@ -0,0 +1,167 @@
+..
+   ****************************************************************************
+    pgRouting Manual
+    Copyright(c) pgRouting Contributors
+
+    This documentation is licensed under a Creative Commons Attribution-Share
+    Alike 3.0 License: http://creativecommons.org/licenses/by-sa/3.0/
+   ****************************************************************************
+
+.. _contraction:
+
+Contraction
+===============================================================================
+
+The documentation provides explanation on graph contraction using small examples.
+
+level1 vertices have only one incoming edge.
+level2 vertices have one incoming edge and one outgoing edge.
+
+For simplicity all the edges in the examples have unit weight.
+
+.. _fig1:
+
+.. figure:: images/twoNodesOneEdge.png
+
+    *Graph 1: Directed, with two nodes and one edge*
+
+    G={V:{1,2},E:{(1,2)}}
+    removed_vertices={}
+
+    level1_vertices={2}
+    level2_vertices={0}
+
+    Cycle 1:
+
+        Level 1:
+
+            level1_vertices is not empty,therefore on level1 contraction, vertex 2 gets deleted.
+
+            removed_vertices={(v,1):{2}}
+
```

Comment:
This notation indicates that:
vertex 2 is removed, and belongs to vertex 1 subgraph